### ISSUE TITLE
Firebase real time database at local emulator  is not showing updated data

### ISSUE THREAD

**Issue by bijaykanshinew** [firebase/firebase-tools#2322]
I have simple code to update data of real time database at  local emaulator

```js
const admin = require('firebase-admin');
const app = admin.initializeApp();
const rdb = app.database();
const gRoomRef = rdb.ref("/chess/groom/2_1")
gRoomRef.update({ move: {}, greeting: {}, chat_msg: {}}).then(() => {
            console.log("success game creation")
        }).catch((error) => {
            console.log("error", error);
        });
```

Neither Code is  throwing any error nor Local emulator (http://localhost:4000/database) is  showing any  data .
It's become bit difficult to test cloud function at my local enviroment.
firebase-tools version => 8.4.1

**Comment by abeisgoat** [comment]
Please follow the template when reporting bugs.

Are you seeing the data written to your production database?

**Comment by bijaykanshinew** [comment]
>Are you seeing the data written to your production database?
No, it's not updating production database

**Comment by bijaykanshinew** [comment]
i have deployed cloud function in production to test code. Code was giving success msg but real time database didn't show any data updated by cloud function. 
Whereas react-native code were able to update the real time database

**Comment by samtstern** [comment]
@bijaykanshinew the code you've shown is not a complete Cloud Function so it's hard to tell what's going on.  I have a feeling you're not waiting for `update` to finish.  Try something like this:

```js
const admin = require('firebase-admin');
const app = admin.initializeApp();
const rdb = app.database();

exports.myFunction = functions.https.onRequest((request, response) => {
    const gRoomRef = rdb.ref("/chess/groom/2_1")

    // The 'return' is really important here otherwise the function will 
    // exit without waiting for the result
    return gRoomRef.update({ move: {}, greeting: {}, chat_msg: {}}).then(() => {
            console.log("success game creation")
        }).catch((error) => {
            console.log("error", error);
        });
});
```

**Comment by bijaykanshinew** [comment]
I tried above code but no luck. Database is not getting updated as you can see in the below screen shot

`exports.myFunction = lib.functions.https.onRequest((request, response) => {
      const gRoomRef = lib.rdb.ref("/chess/groom/2_1")
  
      // The 'return' is really important here otherwise the function will 
      // exit without waiting for the result
      return gRoomRef.update({ move: {}, greeting: {}, chat_msg: {}}).then(() => {
              console.log("success game creation");
              response.json({result: `success`});
              return 1;
          }).catch((error) => {
              console.log("error", error);
          });
  });`
screen shot
https://ibb.co/DkPj2HD

**Comment by samtstern** [comment]
@bijaykanshinew thanks for trying that.  However a screenshot is really not enough information for debugging.

In the emulator UI I notice you have multiple databases, are you sure you're writing to the right one?

Also are you geting any logs from your function?  Any errors?

Can you print out the full path of `gRoomRef` and make sure it's pointing to localhost and to the proper database?

**Comment by SkrewEverything** [comment]
@bijaykanshinew @samtstern  I encountered similar bug today but with the firestore emulator. When I try to add new data to firestore, the data is being written without any error but when I tried to see the data in the emulator UI, it was empty. But when I query it from the firebase functions, I'm getting back my data.

After fiddling around, I inserted some random data through the emulator UI in the firestore and after that I'm able to see the changes in the firestore UI as expected.

So, I advice @bijaykanshinew to insert some data and query for that through functions. If you are getting the data you have inserted,  then try to insert random value into the real time database through emulator UI and then try to run your functions to insert new data and then check whether it showed up or not in the emulator UI.

**My environment:** 

**Firebase**: 8.4.2
**OS**: macOS Catalina 10.15.5
